models/repo_list.go

```diff
@@ -114,51 +114,91 @@ type SearchRepoOptions struct {
 
 // SearchRepositoryByName takes keyword and part of repository name to search,
 // it returns results in given range and number of total results.
-func SearchRepositoryByName(opts *SearchRepoOptions) (repos RepositoryList, count int64, err error) {
-	var cond = builder.NewCond()
+func SearchRepositoryByName(opts *SearchRepoOptions) (repos RepositoryList, _ int64, _ error) {
+	// Set owner of searched repositories if present in options
+	searchOwnerID := opts.OwnerID
+
+	// Set owner ID to Searcher ID if Owner is not filled
+	if opts.OwnerID <= 0 && opts.Searcher != nil {
+		searchOwnerID = opts.Searcher.ID
+	}
+	// Check if user with Owner ID exists
```

Comment:
It's about opinion. Why would you do search if you know, there is no possible results? Also why not answer to searcher that he is using bad ID instead of telling him, this ID is OK but has no repositories.